blocks/main_loop.py

```diff
@@ -156,16 +159,17 @@ def _check_finish_training(self):
         # the iteration the corresponding log record can be found only in
         # the previous row.
         if (self.log.current_row.training_finish_requested or
-                self.log.current_row.keyboard_interrupt_received or
-                self.log.previous_row.keyboard_interrupt_received):
+                self.log.current_row.interrupt_received or
+                self.log.previous_row.interrupt_received):
             raise TrainingFinish
 
-    def _handle_keyboard_interrupt(self, signal_number, frame):
+    def _handle_interrupt(self, signal_number, frame):
         # After receiving a first keyboard interrupt signal,
         # ignore all following ones.
```

Comment:
Actually, you can just remove the old one and rename the current one, that gives you zero chance of ruining a good copy with a bad copy.